### DISCUSSION TITLE
Threading a run with child runs that are executed in another thread / process

### DISCUSSION THREAD

**Discussion by vasanth-asokan** [langchain-ai/langsmith-sdk#1021] [Q&A] [answered]
The [Working with Runs docs](https://docs.smith.langchain.com/old/cookbook/tracing-examples/traceable#working-with-runs) mention 
>Continue a trace in other functions that may be called within a separate thread or process pool (or even on separate machines)

Is there an example of how to do this? I am not using langchain, but using langsmith. I have a function, say `parallel_retriever` decorated with `@traceable` and it executes other functions decorated with @traceable in parallel threads. How can I make sure all the runs are nested under the `parallel_retriever` trace?

I've tried the approach below, but child runs are not nested:

```
def parent_run_method(run_tree: RunTree):
    ... schedule calls to parallel_runs(parent_run_tree=run_tree)

def parallel_runs(parent_run_tree: RunTree):
        with langsmith.tracing_context(parent=parent_run_tree):
            ... child run logic ... 
```

Appreciate for any pointers or help!

**Comment by hinthornw** [comment, marked as answer]
Is this helpful? https://docs.smith.langchain.com/how_to_guides/tracing/nest_traces#context-propagation-using-threading

**Reply by vasanth-asokan** [reply]
Thank you so much! Both methods worked! I must have missed this in the docs.

**Reply by hinthornw** [reply]
yay glad you're unblocked :)
